internal/ingress/controller/parser/parser.go

```diff
@@ -1086,6 +1086,35 @@ func validateProtocol(protocol string) bool {
 	return match
 }
 
+// useSSLProtocol updates the protocol of the route to either https or grpcs, or https and grpcs
+func useSSLProtocol(route *kong.Route) {
+	var http, grpc bool
+	var prots []*string
+
+	for _, val := range route.Protocols {
+
+		if strings.Contains(*val, "grpc") {
+			grpc = true
+		}
+
+		if strings.Contains(*val, "http") {
+			http = true
+		}
+	}
+
+	if grpc {
+		prots = append(prots, kong.String("grpcs"))
+	}
+	if http {
+		prots = append(prots, kong.String("https"))
+	}
```

Comment:
Tests for these are missing. Please add them.
Also the current test-suite seems to be failing as well.